CONTRIBUTING.md

```diff
@@ -0,0 +1,55 @@
+# Contributing to `nexus-sdk`
+
+Thank you for considering contributing to this project!
+The following guidelines help ensure a smooth process for everyone.
+
+## Code of Conduct
+
+This project adheres to a [Code of Conduct]. By participating, you are expected to uphold this code.
+
+## CLA or DCO requirement
+
+By contributing to `nexus-sdk`, you agree that ... TBD
+
+
+## How to Contribute
+
+1. **Fork the repository** and create your branch from `main`.
+2. **Follow the coding style** used in the project.
+3. **Write clear, concise commit messages.**
+4. **Add tests** for any new functionality.
+5. **Ensure all tests pass** before submitting a pull request.
+6. **Ensure `rustfmt` and `clippy` are happy** before submitting a pull request.
+7. **Do not use unsafe Rust code**.
+
+## Reporting Issues
+
+- Use the [Issue Tracker] to report bugs or suggest enhancements.
+- Check if the issue already exists before submitting.
+- Provide steps to reproduce, expected vs. actual behavior, and relevant logs or screenshots.
+
+## Pull Request Process
+
+- Open a pull request with a clear description of what it does and why.
+- Link to the issue it fixes (if applicable).
+- Ensure your branch is up to date with `main`.
+- A maintainer will review and may request changes.
+
+Your PR will be reviewed by maintainers and may be merged after approvals.
+
+## Writing commit messages
+- Use the present tense ("Add feature" not "Added feature")
+- Use imperative tone ("Add feature" not "Adds feature" or "Adding feature")
+- Reference a ticket if applicable.
+- Add more context and an explanation for the contribution.
```

Comment:
Do we want this in the commit message or as a comment in the PR?

Reply:
I have a very specific opinions about this. My answer is it has to be in the commit message, and optionally in the PR. The reason is that if it is in the commit message then it's part of the history of the project. If it is in the PR, then it is not, which I find as a disadvantage.

Reply:
In an ideal world, all PR details should be part of the git repo, right now they live on GitHub, so they are _detached_ from the repository, in a very precise way.

If a service comes along that unifies the PRs with the repo history, they can have my $$.

Reply:
I did not consider this part. Feel free to resolve then. But, does that mean we have to change how we do things internally as well? (I'm not sure anyone is adding more context in the commit messages atm)

Reply:
I try to show a better way and I give hints, and in a world where I do not want to impose 1000 rules, it's all about individual awareness and understanding :)

All good for now. 99.9% of the teams out there put the emphasis on the PR description and not the commit message anyway.